integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go

```diff
@@ -157,178 +162,467 @@ type keysHolder struct {
 	ethAddress common.Address
 }
 
-type testConfig struct {
-	isMintBurnOnMvX bool
-	isNativeOnMvx   bool
-	isMintBurnOnEth bool
-	isNativeOnEth   bool
+func TestRelayersShouldExecuteTransfers(t *testing.T) {
+	t.Run("ETH->MVX and back, ethNative = true, ethMintBurn = false, mvxNative = false, mvxMintBurn = true", testRelayersShouldExecuteTransfersEthToMVX(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexTrue,
+			mvxHexIsNative:   hexFalse,
+			ethIsMintBurn:    false,
+			ethIsNative:      true,
+		},
+	))
+	t.Run("MVX->ETH, ethNative = false, ethMintBurn = true, mvxNative = true, mvxMintBurn = false", testRelayersShouldExecuteTransfersMVXToETH(
+		argSimulatedSetup{
+			mvxHexIsMintBurn: hexFalse,
+			mvxHexIsNative:   hexTrue,
+			ethIsMintBurn:    true,
+			ethIsNative:      false,
+		},
+	))
 }
 
-func TestTransfersBothWaysWithChainSimulator(t *testing.T) {
-	t.Run("Eth: native, MvX: mint & burn", func(t *testing.T) {
-		cfg := testConfig{
-			isMintBurnOnMvX: true,
-			isNativeOnMvx:   false,
-			isMintBurnOnEth: false,
-			isNativeOnEth:   true,
+func testRelayersShouldExecuteTransfersEthToMVX(argsSimulatedSetup argSimulatedSetup) func(t *testing.T) {
+	return func(t *testing.T) {
+		defer func() {
+			r := recover()
+			if r != nil {
+				require.Fail(t, "should have not panicked")
+			}
+		}()
+
+		argsSimulatedSetup.t = t
+		testSetup := prepareSimulatedSetup(argsSimulatedSetup)
+		defer testSetup.close(t)
```

Comment:
to avoid passing the `t` instance here, could have also stored the `t` instance of type `*testing.T` inside the `testSetup` struct.